connector/spanmetricsconnector/connector_test.go

```diff
@@ -769,10 +827,57 @@ func BenchmarkConnectorConsumeTraces(b *testing.B) {
 	}
 }
 
-func newConnectorImp(t *testing.T, mcon consumer.Metrics, defaultNullValue *string, histogramConfig func() HistogramConfig, temporality string, logger *zap.Logger, ticker *clock.Ticker) *connectorImp {
+func TestExcludeDimensionsConsumeTraces(t *testing.T) {
+	mcon := &mocks.MetricsConsumer{}
+	mcon.On("ConsumeMetrics", mock.Anything, mock.Anything).Return(nil)
+	excludeDimensions := []string{"span.kind", "span.name", "totallyWrongNameDoesNotAffectAnything"}
+	p := newConnectorImp(t, mcon, stringp("defaultNullValue"), explicitHistogramsConfig, cumulative, zaptest.NewLogger(t), nil, excludeDimensions...)
+	traces := buildSampleTrace()
+
+	// Test
+	ctx := metadata.NewIncomingContext(context.Background(), nil)
+
+	p.ConsumeTraces(ctx, traces)
+	metrics := p.buildMetrics()
+
+	for i := 0; i < metrics.ResourceMetrics().Len(); i++ {
+		rm := metrics.ResourceMetrics().At(i)
+		ilm := rm.ScopeMetrics()
+		// Checking all metrics, naming notice: ilmC/mC - C here is for Counter.
+		for ilmC := 0; ilmC < ilm.Len(); ilmC++ {
+			m := ilm.At(ilmC).Metrics()
+			for mC := 0; mC < m.Len(); mC++ {
+				metric := m.At(mC)
+				// We check only sum and histogram metrics here, because for now only they are present in this module.
+
+				if metric.Type() == pmetric.MetricTypeExponentialHistogram || metric.Type() == pmetric.MetricTypeHistogram {
```

Comment:
I will cause panic only in case when you are trying to get 
`metric.Sum().DataPoints()` from histogram metrics. 
If you will get DataPoints() from pmetric.Sum then everything will be fine. 

<img width="960" alt="image" src="https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/1440217/211f26be-671d-468a-940c-0ed1913ccc2e">